api/src/pcapi/routes/backoffice_v3/bookings/helpers.py

```diff
@@ -0,0 +1,93 @@
+import datetime
+from typing import Union
+
+from flask_sqlalchemy import BaseQuery
+
+from pcapi.core.bookings import models as bookings_models
+from pcapi.core.categories import subcategories_v2
+from pcapi.core.educational import models as educational_models
+from pcapi.core.finance import models as finance_models
+from pcapi.core.users import models as users_models
+from pcapi.routes.backoffice_v3.bookings.form import BaseBookingListForm
+from pcapi.utils import date as date_utils
+from pcapi.utils import email as email_utils
+
+
+def get_bookings(  # type: ignore [no-untyped-def]
+    base_query: BaseQuery,
+    form: BaseBookingListForm,
+    stock,
+    booking,
+    offer,
+    search_by_email: bool = False,
+    id_filters: list = None,
+    name_filters: list = None,
+    or_filters: list = None,
```

Comment:
Ce `type: ignore` me gêne car on ne souhaite pas laisser de paramètres non typés.
Peut-être quelque chose comme ça pour commencer : 
```suggestion
def get_bookings(
    base_query: BaseQuery,
    form: BaseBookingListForm,
    stock_class: typing.Type[educational_models.CollectiveStock | offers_models.Stock],
    booking_class: typing.Type[educational_models.CollectiveBooking | bookings_models.Booking],
    offer_class: typing.Type[offers_api.AnyOffer],
    search_by_email: bool = False,
    id_filters: list | None = None,
    name_filters: list | None = None,
    or_filters: list | None = None,
```